pkg/kubelet/kubelet_pods.go

```diff
@@ -736,12 +736,8 @@ func (kl *Kubelet) podIsTerminated(pod *v1.Pod) bool {
 // OkToDeletePod returns true if all required node-level resources that a pod was consuming have
 // been reclaimed by the kubelet.  Reclaiming resources is a prerequisite to deleting a pod from the
 // API server.
-func (kl *Kubelet) OkToDeletePod(pod *v1.Pod) bool {
-	if pod.DeletionTimestamp == nil {
-		// We shouldnt delete pods whose DeletionTimestamp is not set
-		return false
-	}
-	if !notRunning(pod.Status.ContainerStatuses) {
+func (kl *Kubelet) OkToDeletePod(pod *v1.Pod, status v1.PodStatus) bool {
+	if !notRunning(status.ContainerStatuses) {
```

Comment:
Is this to avoid rountripping via the api server? if yes, add comments explaining it.